resources/user.rb

```diff
@@ -0,0 +1,57 @@
+resource_name :user_ulimit
+
+property :username, String, name_property: true
+property :filename, String, default: lazy { |r| r.username == '*' ? '00_all_limits' : "#{r.username}_limits" }
+property :filehandle_limit, [String, Integer]
+property :filehandle_soft_limit, [String, Integer]
+property :filehandle_hard_limit, [String, Integer]
+property :process_limit, [String, Integer]
+property :process_soft_limit, [String, Integer]
+property :process_hard_limit, [String, Integer]
+property :memory_limit, [String, Integer]
+property :core_limit, [String, Integer]
+property :core_soft_limit, [String, Integer]
+property :core_hard_limit, [String, Integer]
+property :stack_limit, [String, Integer]
+property :stack_soft_limit, [String, Integer]
+property :stack_hard_limit, [String, Integer]
+property :rtprio_limit, [String, Integer]
+property :rtprio_soft_limit, [String, Integer]
+property :rtprio_hard_limit, [String, Integer]
+property :virt_limit, [String, Integer]
+
+action :create do
+  new_resource.filename = "#{new_resource.filename}.conf" unless new_resource.filename.include?('.conf')
```

Comment:
that looks like a coercion

Reply:
I added some specs so I can refactor this and not break the world. I'm going to get this part in first though.